cpp/arcticdb/async/async_store.hpp

```diff
@@ -280,36 +280,35 @@ class AsyncStore : public Store {
         return res;
     }
 
-    std::vector<VariantKey> batch_read_compressed(
+    folly::Future<std::vector<VariantKey>> batch_read_compressed(
```

Comment:
Please mark all `Future` returning functions with `[[nodiscard]]`

Reply:
I don't see a consistent usage of this for folly over the code. I will leave it open to discussion

Reply:
> I don't see a consistent usage of this for folly over the code.

It would be hard doing any improvements if we have to find consistent usage for feature x in order to add it to the code. With regard to the nodiscard annotation I'd go even further and make the warning into an error. Discarding the result of a function rarely makes sense and is usually a sign of a bug.